### PR TITLE
Add Dockerfile for image based on local source copy.

### PR DESCRIPTION
**Pull request by russellb**
The existing Dockerfiles build an image based on the latest master
branch of flowdec.  This variation lets you build a docker image based
on the current local copy of the source.

### CODE REVIEW

**Comment by eric-czech** [comment]
Hey @russellb , the editable pip install makes a lot of sense but I'm wondering if you often prefer to make changes to source in the container or if it makes more sense to do something like this:

```
# Local copy of repo to do development on
LOCAL_REPO_DIR=$HOME/repos/flowdec

# Linked volume in container 
CONTAINER_REPO_DIR=/repos/flowdec

nvidia-docker run -ti -p 8888:8888 -v $LOCAL_REPO_DIR:$CONTAINER_REPO_DIR flowdec
```

Then in the Dockerfile:

```
RUN cd /repos/flowdec/python && pip install -e .
```

Do you think it would be better this way so that you can edit the repo in a local directory (i.e. outside of the container) or do you find that there are advantages to doing the development within the container?  I've been something like this lately instead so that I can point something like PyCharm at a repo on my host and know that changes will be available in the container right away, and that the copy of the repo I'm working on will be available when the container is down.  

Let me know if that's a workflow you'd be interested in or if you prefer it the other way.  I don't have a strong opinion regardless but I thought I'd probe to see if there are some interesting pros/cons of the two approaches.

**Comment by russellb** [comment]
I pushed an update.  It's mostly doc updates to reflect using a bind mount for even more convenient editing.  The one caveat is that a new image will need to be built or "pip install -e" will need to be run again if the structure of the source tree changes.

File: docker/Dockerfile.devel [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+# This is most helpful when used in combination with a bind mount of
+# your local source tree on top of this location, so you can make edits
+# as usual, and not within the running container.
+#
+# Note that if you change the structure of the project in any way,
+# like adding/removing/renaming/moving files, you will either need to
+# rebuild the image, or run the "pip install" command again inside the
+# container.
+#
+COPY . /repos/flowdec/.
```
Comment (eric-czech) [inline docker/Dockerfile.devel]: Hey @russellb , does this COPY work if the bind mount is specified linking the source and destination?  Can't say I've ever tried that before but would be cool if it does.

Comment (eric-czech) [reply docker/Dockerfile.devel]: Ah that makes sense, good to know!

**Pull request merged**